packages/blocks/src/store/selectors.js

```diff
@@ -643,6 +643,26 @@ export function hasBlockSupport( state, nameOrType, feature, defaultSupports ) {
 	return !! getBlockSupport( state, nameOrType, feature, defaultSupports );
 }
 
+export function hasBlockMetadataSupport(
```

Comment:
If the attribute and block supports are marked as experimental, then the selector name should also follow and be considered experimental.

Although, I wonder whether we could make a special case for `hasBlockSupport` and `getBlockSupport` and cover it there instead, for example:

```js
import { store as blocksStore } from `@wordpress/blocks`;
import { select } from `@wordpress/data`;

select( blocksStore ).hasBlockSupport( 'metadata.name' );
```

or through the existing public API from `@wordpress/blocks`:

```js
import { hasBlockSupports } from `@wordpress/blocks`;

hasBlockSupports( 'metadata.name' );
```

Reply:
As of https://github.com/WordPress/gutenberg/pull/42605/commits/2c9637984fc60bb5f2ed52518f5584d87a0aac00 the selector is now marked as `__experimental`.

Happy to explore updates to `hasBlockSupport` but I didn't like having to make that function "know" about the shape of attributes being passed in order have any special rules to handle metadata. Thus why I went for a new selector.